xds/src/main/java/com/linecorp/armeria/xds/XdsResponseHandler.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2023 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.xds;
+
+import com.google.protobuf.Message;
+
+import io.envoyproxy.envoy.service.discovery.v3.DiscoveryResponse;
+
+/**
+ * Handles callbacks for {@link SotwXdsStream}.
+ * Note that it is important that exceptions are not raised from the callback.
+ * Otherwise, the infinite loop will break as a discovery request is not sent.
+ */
+interface XdsResponseHandler {
+
+    <T extends Message> void handleResponse(
+            ResourceParser<T> resourceParser, DiscoveryResponse value, SotwXdsStream sender);
+
+    void handleReset(XdsStream sender);
```

Comment:
Could you explain the definition of reset? 🤔

Reply:
Whenever the stream thinks its `SubscriberStorage` may not be synchronized with the control plane, then a `reset` is done. This can be the case if a stream is closed, or if a stream is doing a cold start.

When a reset is performed, all of the currently watched types and resources are sent to the control plane server. Note that this is a package-private interface, and is exposed for testing purposes.

ref: https://github.com/line/armeria/blob/90a93499de29ffac1b27084aba07eeead3c716e7/xds/src/main/java/com/linecorp/armeria/xds/DefaultResponseHandler.java#L84-L86